### PR TITLE
Add autocleaning for optional workload templates during sync.

### PR DESCRIPTION
**Pull request by AR-May**
Delivers issue #2427 .
The changes add optional workload templates' removal when they are no longer returned by the SDK template library.

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs [hunk 1]
```diff
@@ -567,10 +570,25 @@ private bool SyncOptionalWorkloads()
                         Installer.InstallPackages(owInstallationRequests);
                     }
 
-                    /* TODO: add comparison of Optional SDK Workload templates from Hive and returned by TemplateLocator
-                     * implement templates removal if they are not returned by API anymore
-                     */
+                    // remove uninstalled Optional SDK Workload packages
+                    List<string> owRemovalRequestsPackageIds = new List<string>();
+                    foreach(IInstallUnitDescriptor descriptor in _settingsLoader.InstallUnitDescriptorCache.Descriptors.Values)
```
Comment (grinrag) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs:575]: We already iterating through the descriptors, you can just add your code there (several lines above)

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs [hunk 2]
```diff
@@ -535,6 +535,7 @@ private bool SyncOptionalWorkloads()
                 {
                     List<InstallationRequest> owInstallationRequests = new List<InstallationRequest>();
                     Dictionary<string, string> owInstalledPkgs = new Dictionary<string, string>();  // packageId -> packageVersion
+                    HashSet<string> owSyncRequestsPackageIds = new HashSet<string>();
```
Comment (bozturkMSFT) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs:538]: HashSet, my favourite... Nice!

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs (hunk 1 above)
Comment (ladipro) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs:575]: super-nit: Missing space after `foreach`.

Comment (AR-May) [reply src/Microsoft.TemplateEngine.Cli/New3Command.cs]: fixed, thanks

Comment (AR-May) [reply src/Microsoft.TemplateEngine.Cli/New3Command.cs]: No, I can't if I want to use a HashSet for sync requests' ids. But I can use <a>owInstalledPkgs.Keys</a> for iterating. Fixed this.

**Pull request merged**